torch_xla/csrc/tensor_impl.cpp

```diff
@@ -120,6 +120,15 @@ c10::SymIntArrayRef XLATensorImpl::sym_sizes_custom() const {
                              sizes.size());
 }
 
+c10::SymInt XLATensorImpl::sym_numel_custom() const {
+  auto sym_sizes = sym_sizes_custom();
+  c10::SymInt prod{1};
+  for (auto s : sym_sizes) {
+    prod *= s;
```

Comment:
what's the behavior here if symint is actually dynamic? Is it going to use upper bound?

Reply:
it will construct an expression. How we evaluate it (using an upper bound or using an exact value will be determined by how this expression would be used)

Reply:
actually right now it will break on real symints tbh 🤡  since we are missing larger parts of LazySymInt implementation but the above is how things should work in the future